### PR TITLE
WIP: mgr/volumes/nfs: minor enhancements

### PR DESCRIPTION
**Pull request by ajarr**
* allow only alphanumeric characters in NFS cluster ID
* list exports in a NFS cluster
* get export in a NFS cluster
Fixes: https://tracker.ceph.com/issues/45744 
Fixes: https://tracker.ceph.com/issues/45741 

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by ajarr** [comment]
Still needs tests.

**Comment by ajarr** [comment]
@batrick @mgfritch 

Couple of questions,
* Any reason to allow only alphanumeric characters in ganesha cluster IDs? Should we also allow additional chars such as period, underscore and hyphen. Patrick, I know that you didn't want slashes in a cluster ID

* __Listing exports__
  * In this version of the PR, I only output the pseudopaths/bindings in a NFS cluster. In the dashboard code, list_exports outputs the entire content of all the exports in a cluster. Should we do that instead? https://github.com/ceph/ceph/blob/master/src/pybind/mgr/dashboard/controllers/nfsganesha.py#L118
  * And do we want to output the export contents in the native ganesha export syntax or in the python dictionary format (as done in the dashboard code)?

**Comment by mgfritch** [comment]
> * Any reason to allow only alphanumeric characters in ganesha cluster IDs? Should we also allow additional chars such as period, underscore and hyphen. Patrick, I know that you didn't want slashes in a cluster ID

any of those chars should be aok for the orchestrator (period, underscore, hyphen etc.), but avoiding the forward slash makes sense for rados ...


>   * And do we want to output the export contents in the native ganesha export syntax or in the python dictionary format (as done in the dashboard code)?

I'd vote for the python format so the cli can be expanded to support things like `--format yaml` or `--format json` etc.

File: src/pybind/mgr/volumes/fs/nfs.py [hunk 1]
```diff
@@ -364,6 +364,26 @@ def delete_all_exports(self, cluster_id):
...
+        #TODO: Check whether cluster exists
+        result = [export.pseudo for export in self.exports[cluster_id]]
+        return 0, json.dumps(result, indent=2), ''
+
+    def get_export(self, cluster_id, pseudo_path):
+        #TODO: Check whether cluster exists
+        export_obj = None
+        for export in self.exports[cluster_id]:
+            if export.pseudo == pseudo_path:
+                export_obj = "export-{}".format(export.export_id)
```
Comment (varshar16) [inline src/pybind/mgr/volumes/fs/nfs.py:377]: Use `_fetch_export()`.
```suggestion
        self.rados_namespace = cluster_id
        export = self._fetch_export(pseudo_path)
        export_obj = "export-{}".format(export.export_id)
```

**Comment by ajarr** [comment]
This PR was based on top of https://github.com/ceph/ceph/pull/35417/
Once that PR includes the three commits in this PR that fix trackers 45744 and 45741. I will abandon this PR.

**Comment by ajarr** [comment]
This PR is subsumed by https://github.com/ceph/ceph/pull/35417

**Pull request closed without merging**